<a href="https://colab.research.google.com/github/JaydaBubel/lab-inferential-statistics-anova/blob/master/lab_inferential_statistics_anova.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Part 1**

In this activity, we will look at another example. Your task is to understand the problem and write down all the steps to set up ANOVA. After the next lesson, we will ask you to solve this problem using Python. Here are the steps that you would need to work on:
- Null hypothesis
- Alternate hypothesis
- Level of significance
- Test statistic
- P-value
- F table

**Context**

Suppose you are working as an analyst in a microprocessor chip manufacturing plant. You have been given the task of analyzing a plasma etching process with respect to changing Power (in Watts) of the plasma beam. Data was collected and provided to you to conduct statistical analysis and check if changing the power of the plasma beam has any effect on the etching rate by the machine. You will conduct ANOVA and check if there is any difference in the mean etching rate for different levels of power. You can find the data anova_lab_data.xlsx file in the files_for_lab folder

- State the null hypothesis: in this case, there is no significant difference in mean etching rate among different power levels- changing the power level of the plasma beam should not affect etching rate.
- State the alternate hypothesis: that there is a significant difference in mean etching rate among different power levels.
- What is the significance level: typical choices for
α are 0.05 (5%) or 0.01 (1%)
- What are the degrees of freedom of model: with three power levels (160 W, 180 W, 200 W), DFm is 2.
- error terms: how many data points left over; all data points added and # of groups subtracted
- total DoF: degrees of freedom model plus error terms


**Part 2**

In this section, use Python to conduct ANOVA.
What conclusions can you draw from the experiment and why?

In [13]:
import pandas as pd
from scipy import stats

In [14]:
data = pd.read_excel('/content/drive/MyDrive/anova_lab_data.xlsx')
print(data)

   Power   Etching Rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55


In [15]:
print(data.columns)

Index(['Power ', 'Etching Rate'], dtype='object')


In [16]:
groups = data['Power ']
etching_rates = data['Etching Rate']
F_statistic, p_value = stats.f_oneway(*[etching_rates[groups == group] for group in groups.unique()])
print("F-statistic:", F_statistic)
print("P-value:", p_value)

F-statistic: 36.87895470100505
P-value: 7.506584272358903e-06


F-statistic: 36.88 - This number checks if there's a sig. difference between etching rates at different power levels. Higher values suggest bigger differences.

P-value: 7.51... - This small number means there's strong evidence that power levels do affect etching rates. If it's smaller than 0.05, it's prob. significant.

In [17]:
alpha = 0.05
if p_value < alpha:
    print("Reject null hypothesis - There is significant difference in mean etching rates.")
else:
    print("Fail to reject null hypothesis - No significant difference in mean etching rates.")

Reject null hypothesis - There is significant difference in mean etching rates.
